helix-view/src/clipboard.rs

```diff
@@ -243,12 +242,34 @@ pub mod provider {
 
         #[cfg(not(any(windows, target_os = "macos")))]
         pub fn is_exit_success(program: &str, args: &[&str]) -> bool {
-            std::process::Command::new(program)
-                .args(args)
-                .output()
-                .ok()
-                .and_then(|out| out.status.success().then(|| ())) // TODO: use then_some when stabilized
-                .is_some()
+            use helix_lsp::block_on;
+            use tokio::process::Command;
+            use tokio::task::block_in_place;
+            use tokio::time::timeout;
+
+            async fn is_exit_success_async(program: &str, args: &[&str]) -> bool {
+                Command::new(program)
+                    .args(args)
+                    .kill_on_drop(true)
+                    .output()
+                    .await
+                    .ok()
+                    .and_then(|out| out.status.success().then(|| ())) // TODO: use then_some when stabilized
```

Comment:
```suggestion
                    .filter(|out| out.status.success())
```

I think this ends up being a little clearer than `then_some` would be

Reply:
That does look better. Even nicer would be replacing the two lines with `is_some_and(|out| out.status.success())`, but who knows when that will stabilize!